runtime/gc_modron_startup/mminit.cpp

```diff
@@ -3051,15 +3051,40 @@ gcReinitializeDefaultsForRestore(J9VMThread* vmThread)
 {
 	MM_GCExtensions* extensions = MM_GCExtensions::getExtensions(vmThread);
 	J9JavaVM *vm = vmThread->javaVM;
-	bool result = true;
 
 	PORT_ACCESS_FROM_JAVAVM(vm);
-
+	OMRPORT_ACCESS_FROM_J9PORT(PORTLIB);
+	/* Note here we update this parameter which represents the machine physical memory,
+	 * but not the original heap geometry from the snapshot run. We just force the heap
+	 * not to go beyond a certain thereshold calculated based on the new (restore) memory
+	 * availalibity through setting SoftMx.
+	 */
+	extensions->usablePhysicalMemory = omrsysinfo_get_addressable_physical_memory();
+	/* We use false here for computeDefaultMaxHeapForJava(), since the restore
+	 * path is only active for releases after Java 8 releases
+	 */
+	uintptr_t candidateSoftMx = extensions->computeDefaultMaxHeapForJava(false);
+	/* we will set softMx value only if maxHeap calculation returned us a smaller 
+	 * value than existing maxHeap or softMx values and a larger value than existing 
+	 * minHeap value as inherited from/established at snapshot run, this max/minHeap
+	 * is checked by j9gc_set_softmx()
+	 */
+	if (extensions->memoryMax > candidateSoftMx) {
+		if ((0 == extensions->softMx) || (extensions->softMx > candidateSoftMx)) {
+			if (1 == j9gc_set_softmx(vm, candidateSoftMx)){
```

Comment:
since the failure message is rather specific, we probably want this check to be specific (against initialMemorySize) to match it

Reply:
also watch spacing (ending '){')